### DISCUSSION TITLE
Apple Silicon Support?

### DISCUSSION THREAD

**Discussion by AaronBeaudoin** [Hammerspoon/hammerspoon#3140] [General]
Hello.

I've got Hammerspoon installed on my new Macbook Pro 14" with an M1 Pro. My main use cases for it was to automate moving screens arounds and rotating them, as well as moving windows around from screen to screen.

I ran into my first issue within a few minutes. I can't rotate screens, and it looks like I'm not the only one with this issue (#3041). It looks like the issue has been around for 3 months now, and nobody's looked at it. I personally don't know enough about Apple APIs (or even Lua for that matter) to implement a fix myself, so I'm kind of just stuck until someone with the experience is able to take a look at it.

Overall though, I guess my question is just regarding Apple Silicon compatibility in general. Is there any sort of compatibility table anywhere which shows which Hammerspoon features have been tested to be working on ARM Macs? I really love the idea of Hammerspoon but I can't get real excited about it if ARM compatibility is just hit or miss. What are your thoughts?

**Comment by latenitefilms** [comment]
We have quite a lot of [CommandPost](http://commandpost.io) (which is a fork of Hammerspoon) users using Monterey on Apple Silicon, and haven't come across any major issues or problems. I'll test out #3041 on my 16-inch AS MacBook Pro later today.

**Comment by cmsj** [comment]
On the whole our ARM compatibility is pretty high, but we don't have an official record of it. I think it would be interesting to have notes in the API docs to indicate whether or not a particular function/method works on Intel/ARM/Both, I'd be very happy to see docstrings patches which add that kind of information.
